CausalPrediction.py

```diff
@@ -0,0 +1,315 @@
+import statsmodels.api as sm
+from doubleml import DoubleMLData, DoubleMLPLR
+from sklearn.ensemble import RandomForestRegressor
+from statsmodels.sandbox.regression.gmm import IV2SLS
+
+from pgmpy.base import DAG, PDAG
+from pgmpy.estimators import PC
+from pgmpy.inference import CausalInference
+from pgmpy.models import LinearGaussianBayesianNetwork as LGBN
+
+
+class CausalGraphWithVariableRoles:
+    """
+    Class for specifying a Causal Graph.
+
+    If `graph` is not provided, a causal discovery algorithm is
+    used to learn the DAG.
+
+    Parameters
+    ----------
+    graph: pgmpy.models.DAG or Causal Discovery Algorithm instance.
+        The DAG representing the causal graph over all relevant variables.
+
+    exposure: str
```

Comment:
exposure and outcome will be roles in the `graph`, after https://github.com/pgmpy/pgmpy/pull/2265